src/SymbolTable.h

```diff
@@ -133,6 +133,19 @@ class SymbolTable {
         return newSymbolOfIndex(symbol);
     }
 
+    /** Finds the index of a symbol in the table, giving an error if it's not found */
+    const size_t lookupExisting(const std::string str) const {
```

Comment:
str can be const ref.

Also, is this method needed? Is it likely to be called with a non-existent string?

Also also, if it is needed, then use const char* for consistency with the other lookup methods.